pandas/indexes/base.py

```diff
@@ -1328,22 +1328,43 @@ def _ensure_compat_concat(indexes):
 
         return indexes
 
-    def take(self, indices, axis=0, allow_fill=True, fill_value=None):
-        """
-        return a new Index of the values selected by the indexer
+    _index_shared_docs['take'] = """
+        return a new Index of the values selected by the indices
 
         For internal compatibility with numpy arrays.
 
-        # filling must always be None/nan here
-        # but is passed thru internally
+        Parameters
+        ----------
+        indices : list
+            Indices to be taken
+        axis : int, optional
+            The axis over which to select values, always 0.
+        allow_fill : bool, default True
```

Comment:
see my comment. I think we should just take this out (well i guess deprecate it, then take it out in a future version). This is really not a user parameter.